packages/wordcount/src/types.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * Possible ways of counting text.
+ */
+export type WPWordCountStrategy =
+	| 'words'
+	| 'characters_excluding_spaces'
+	| 'characters_including_spaces';
+
+/**
+ * L10n settings for word counting.
+ */
+export interface WPWordCountL10n {
+	/**
+	 * The type of count to perform.
+	 */
+	type?: WPWordCountStrategy;
+
+	/**
+	 * Array of shortcode names to be removed during counting.
+	 */
+	shortcodes?: string[];
+}
+
+/**
+ * Base settings fields that can be configured by users.
+ */
+export interface WPWordCountUserSettings {
+	/**
+	 * Regular expressions for various text processing operations.
+	 */
+	HTMLRegExp?: RegExp;
+	HTMLcommentRegExp?: RegExp;
+	spaceRegExp?: RegExp;
+	HTMLEntityRegExp?: RegExp;
+	connectorRegExp?: RegExp;
+	removeRegExp?: RegExp;
+	astralRegExp?: RegExp;
+	wordsRegExp?: RegExp;
+	characters_excluding_spacesRegExp?: RegExp;
+	characters_including_spacesRegExp?: RegExp;
+
+	/**
+	 * Localization settings.
+	 */
+	l10n?: WPWordCountL10n;
+}
```

Comment:
We can re-use the already defined type to retain the documentation for the properties

```ts
export type WPWordCountUserSettings = Partial<
	Omit< WPWordCountSettings, 'type' | 'shortcodes' >
>;
```